tests/conftest.py

```diff
@@ -21,3 +27,34 @@ def app(django_app):
 @pytest.fixture
 def csrf_exempt_django_app(django_app_factory):
     return django_app_factory(csrf_checks=False)
+
+
+def _run_sql(settings, sql):
+    conn = psycopg2.connect(
+        dbname='postgres',
+        host=settings['HOST'],
+        port=settings['PORT'],
+        user=settings['USER'],
+        password=settings['PASSWORD'],
+    )
+    conn.set_isolation_level(ISOLATION_LEVEL_AUTOCOMMIT)
+    cur = conn.cursor()
+    cur.execute(sql)
+    conn.close()
+
+
+@pytest.yield_fixture(scope='session')
+def django_db_setup(django_db_blocker):
+    from django.conf import settings
+    test_db_name = 'test_adp_dev'
+    settings.DATABASES['default']['NAME'] = test_db_name
+    run_sql = partial(_run_sql, settings.DATABASES['default'])
+    run_sql(f'DROP DATABASE IF EXISTS {test_db_name}')
+    run_sql(f'CREATE DATABASE {test_db_name}')
+    with open(settings.BASE_DB_PATH, 'r') as file:
+        sqlFile = file.read()
+        with django_db_blocker.unblock():
+            with connection.cursor() as cursor:
+                cursor.execute(sqlFile)
+            call_command('migrate')
+        file.close()
```

Comment:
Could you move this out of `conftest.py` into docker-compose.yml, similarly like it was done with MySQL?

See https://github.com/docker-library/docs/blob/master/postgres/README.md#initialization-scripts

Reply:
Would need some guidance on how to do that, as per recent commits I have tried it, but nothing seems to work if I have my original code commented out.

Reply:
Maybe this would work?

```sh
# Load initial database schema
psql -h localhost -U adp -d adp-dev < resources/adp-pg.sql

# Run migrations
poetry run python manage.py migrate

# Rename database to match test database name
psql -h localhost -U adp postgres -c 'alter database "adp-dev" rename to "test_adp-dev";'

# And run tests with `--reuse-db` on original database with migrations applied on top of it
poetry run pytest -vvxra --tb=short --reuse-db tests
```

We need to only run these commands on CI, to reproduce exactly the same conditions as in production setting. And by doing that I already found an issue:

```
psycopg2.errors.UndefinedColumn: column usecase_dataset_ids.id does not exist
```

Here, `UsercaseDatasetIds` is incorrectly defined, maybe it is possible to fix it by defining a proper [ManytoManyField.through](https://docs.djangoproject.com/en/3.2/ref/models/fields/#django.db.models.ManyToManyField.through), but it looks, that this too requires `id` field, if this is the case, the probably only way to fix it is through migrations.

Reply:
Since we are running migrations manually, i added --no-migrations and --reuse-db to the tests.